scrapinghub/client.py

```diff
@@ -237,20 +237,24 @@ def purge(self):
         self.metadata.expire()
 
     def start(self, **params):
-        return self._project.jobq.start(self, **params)
+        return self.update(state='running', **params)
+
+    def finish(self, **params):
+        return self.update(state='finished', **params)
+
+    def delete(self, **params):
+        return self.update(state='deleted', **params)
 
     def update(self, **params):
-        return self._project.jobq.update(self, **params)
+        try:
+            job = next(self._project.jobq.update(self, **params))
+            return job['prevstate']
+        except StopIteration:
+            raise ValueError("Job {} doesn't exist".format(self.key))
```

Comment:
User can pass any non-existing job key, for me `ValueError` or `KeyError` should fit, do you have better idea in mind?